pkg/util/k8sops/peerpod.go

```
@@ -0,0 +1,125 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8sops
+
+import (
+	"context"
+	"log"
+	"os"
+
+	peerPodV1alpha1 "github.com/confidential-containers/cloud-api-adaptor/peer-pod-controller/api/v1alpha1"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+)
+
+var logger = log.New(log.Writer(), "[util/k8sops] ", log.LstdFlags|log.Lmsgprefix)
+var ppFinalizer string = "peer.pod/finalizer"
+
+type PeerPodService struct {
+	client        *kubernetes.Clientset
+	uclient       *rest.RESTClient // use generated client instaed
+	cloudProvider string
+}
+
+func NewPeerPodService() *PeerPodService {
+	cloudProvider := os.Getenv("CLOUD_PROVIDER") // TODO: don't get from env var directly
+	if cloudProvider == "" {
+		logger.Printf("NewPeerPodService: failed to get cloudProvider")
+		return nil
```

Can we return an error explicitly like this?

```
		return nil, errors.New("NewPeerPodService: failed to get cloudProvider")
```